### PR TITLE
ledger: Replace saturating_add_assign with Saturating<T>

### PR DESCRIPTION
**Pull request by roryharr**
#### Problem
https://github.com/anza-xyz/agave/issues/502

I'm not clear if this cleanup is still wanted, but I asked on the issue and didn't get a response. 

Will work on the rest if this is still a useful cleanup.  

#### Summary of Changes
Remove instances of saturating_add_assign from ledger

Fixes #

### CODE REVIEW

File: ledger/src/blockstore_processor.rs [hunk 1]
```diff
@@ -1339,11 +1339,8 @@ impl ThreadExecuteTimings {
 
     pub fn accumulate(&mut self, other: &ThreadExecuteTimings) {
         self.execute_timings.accumulate(&other.execute_timings);
-        saturating_add_assign!(self.total_thread_us, other.total_thread_us);
-        saturating_add_assign!(
-            self.total_transactions_executed,
-            other.total_transactions_executed
-        );
+        self.total_thread_us += other.total_thread_us;
+        self.total_transactions_executed += other.total_transactions_executed;
```
Comment (buffalojoec) [inline ledger/src/blockstore_processor.rs:1343]: This isn't saturating anymore.

**Review by buffalojoec — COMMENTED** [review]
I'm going to be honest with you, I don't think this really adds a lot of value, so I'm inclined to not include this change. If anyone else feels differently, feel free to weigh in, but wanted to get you prompt feedback to save you time @roryharr.

Thanks as always for contributing, and if you spot something else you want to work on, by all means!

File: ledger/src/blockstore_processor.rs (hunk 1 above)
Comment (steviez) [reply ledger/src/blockstore_processor.rs:1343]: It is still saturating math as the types were declared `Saturating` above:
```rust
    pub total_thread_us: Saturating<u64>,
    pub total_transactions_executed: Saturating<u64>,
```
https://doc.rust-lang.org/beta/std/num/struct.Saturating.html

**Review by steviez — COMMENTED** [review]
> I'm not clear if this cleanup is still wanted, but I asked on the issue and didn't get a response.
> 
> Will work on the rest if this is still a useful cleanup.

Apologies about that. I'll review this as I still think it would be nice to remove the workaround now that the behavior we want is baked into `std::num`

**Review by steviez — APPROVED** [review]
LGTM - I hit the wrong button earlier so CI is running right now; will merge after it completes (assuming no issues)

**Comment by roryharr** [comment]
Great. I'll start working on another. I will try and keep the reviews relatively small (larger than this though!) to make it fast and easy to review.

**Pull request merged**